src/components/CheckoutDrawer/index.jsx

```diff
@@ -0,0 +1,155 @@
+import React from "react";
+import { Drawer, Form, Input, Button } from "antd";
+import { CardElement } from "@stripe/react-stripe-js";
+import { useTranslation } from "react-i18next";
+import firebase from "../../firebaseConfig";
+
+const db = firebase.firestore();
+
+const CheckoutDrawer = ({
+  drawerVisible,
+  setDrawerVisible,
+  cart,
+  isTesting,
+}) => {
+  const { t } = useTranslation();
+  const onClose = () => {
+    setDrawerVisible(false);
+  };
+
+  const layout = {
+    labelCol: {
+      span: 4,
+    },
+    wrapperCol: {
+      span: 20,
+    },
+  };
+  const tailLayout = {
+    wrapperCol: {
+      offset: 2,
+      span: 16,
+    },
+  };
+
+  const options = {
+    style: {
+      base: {
+        fontSize: "16px",
+        color: "#424770",
+        letterSpacing: "0.025em",
+        fontFamily: "Source Code Pro, monospace",
+        "::placeholder": {
+          color: "#aab7c4",
+        },
+      },
+      invalid: {
+        color: "#9e2146",
+      },
+    },
+  };
+
+  const handleSubmit = () => {
+    const todaysDate = new Date();
+    const todaysMonthIndex = todaysDate.getMonth();
+    cart.forEach((coupon) => {
+      db.collection("sampleSoldCoupons")
+        .doc(coupon.restaurantId)
+        .set({
+          compName: coupon.organizationName,
+        })
+        .then(() => {
+          for (let i = 0; i < coupon.quantity; i++) {
+            db.collection("sampleSoldCoupons")
+              .doc(coupon.restaurantId)
+              .collection("coupons")
+              .add({
+                amount: coupon.amount + "₺",
+                isActive: true,
+              });
+          }
+        })
+        .then(() => {
+          const couponStatistics = [];
+          db.collection("soldCoupons")
+            .doc(coupon.restaurantId)
+            .get()
+            .then((doc) => couponStatistics.push(doc.data()))
+            .then(() => {
+              console.log("asd", couponStatistics[0]);
+              const sinlgePriceStatistics =
+                couponStatistics[0][coupon.amount + "₺"];
+              sinlgePriceStatistics[todaysMonthIndex] += coupon.quantity;
+              db.collection("soldCoupons")
+                .doc(coupon.restaurantId)
+                .update({
+                  [coupon.amount + "₺"]: sinlgePriceStatistics,
+                });
+              console.log(sinlgePriceStatistics);
+            });
+        });
+    });
+  };
+
+  return (
+    <>
+      <Drawer
+        width={600}
+        title="Basic Drawer"
+        placement="right"
+        onClose={onClose}
+        visible={drawerVisible}
+      >
+        <Form {...layout} onFinish={handleSubmit}>
+          <Form.Item
```

Comment:
why isn't these form items controlled by states ? won't the user input be used ?

Reply:
These forms are there now just for show since we weren't able to implement the payment. However, buying coupons will update the restaurant's statistics.